### PR TITLE
[WIP] Logical ANDs for --groups

### PR DESCRIPTION
**Pull request by faso**
Related to #9 

Haven't gotten very deeply into the codebase yet, but this seems to do the trick in simple scenarios.

@vptes1 could you please review this and tell me if there are any massive misunderstandings on my part? And do I need to consider any other places these changes might affect?

If not, I will go on to the testing and commit cleanup.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -130,7 +130,9 @@ function processFlag(name, value) {
                 if(!value) {
                     utils.error(`groups flag must list the groups to run, separated by commas`);
                 }
-                runner.groups = value.split(/\s*\,\s*/);
+                // runner.groups used to be ['a', 'b']
+                // now it's [ ['a'], ['b', 'c'] ] === --groups=a,b+c === A or (B and C)
+                runner.groups = value.split(/\s*\,\s*/).map(group => group.split('+'));
```
Comment (vptes1) [inline src/cli.js]: Maybe `group.split(/\s*\+\s*/)` instead, so you can put whitespace around `+`es

File: src/cli.js [hunk 2]
```diff
@@ -130,7 +130,9 @@ function processFlag(name, value) {
                 if(!value) {
                     utils.error(`groups flag must list the groups to run, separated by commas`);
                 }
-                runner.groups = value.split(/\s*\,\s*/);
+                // runner.groups used to be ['a', 'b']
+                // now it's [ ['a'], ['b', 'c'] ] === --groups=a,b+c === A or (B and C)
```
Comment (vptes1) [inline src/cli.js]: I'd move the above 2 comments to where `this.groups` is defined in `runner.js`

**Comment by vptes1** [comment]
Looks really good so far! Don't forget to add unit tests to `tests/core/runner.js` :)

**Comment by faso** [comment]
Let me know if you'd like more tests or some other changes! Also, is there a way to contribute to the docs on the website?

**Comment by vptes1** [comment]
For now, send me what you'd like added to the site. I'll probably upload the site's code into another repo soon.

**Pull request merged**